### DISCUSSION TITLE
Checking for open sessions and re-using open session to send multiple messages

### DISCUSSION THREAD

**Discussion by cseguinlz** [3mora2/WPP_Whatsapp#32] [Q&A]
So right now I basically open a new browser session with each message I send...

The challenge is that I'm integrating WPP_Whatsapp in a FastApi project and is not working. So I'm treating it as a script, with its own  run by a subprocess, called from a router. That works fine, but everytime the process is run I need:
1. to check if there's a session open,
2. if there is, somehow recover current `client` instance and send the message.

how do you check for open sessions and how do you use them to send new messages to it?

Been trying I few things but I don't see how to make it work, I can save the state in file or even DB, but still, don't see how to recover current `client`

Any ideas?

**Comment by 3mora2** [comment]
[wppconnect-serve](https://github.com/wppconnect-team/wppconnect-server) is already exist to handle multiple session.
But you can [try multi_session.py](https://github.com/3mora2/WPP_Whatsapp/blob/main/examples/multi_session.py), 
Make global var to session
Everytime check if session is found
And check status
